### PR TITLE
drivers: nxp_enet: Add ability to use fused mac on some platforms

### PR DESCRIPTION
**Pull request by decsny**
Add ability to use fused mac on RT10xx and RT11xx and redocument the nxp,unique-mac property to be more clear of what it is

### CODE REVIEW

**Review by danieldegrasse — COMMENTED** [review]
Changes look good, although it seems there are some CI failures on the IP K66 board

File: drivers/ethernet/nxp_enet/eth_nxp_enet.c [hunk 1]
```diff
@@ -566,18 +572,48 @@ static void eth_nxp_enet_isr(const struct device *dev)
...
 	mac_addr[3] = FIELD_GET(0xFF0000, id);
 	mac_addr[4] = FIELD_GET(0x00FF00, id);
 	mac_addr[5] = FIELD_GET(0x0000FF, id);
 }
 
+#ifdef CONFIG_SOC_FAMILY_NXP_IMXRT
+#include <fsl_ocotp.h>
+#endif
+
+static inline void nxp_enet_fused_mac(uint8_t *mac_addr)
```
Comment (pdgendt) [inline drivers/ethernet/nxp_enet/eth_nxp_enet.c:593]: Can you return `-ENOTSUP` for unsupported SoCs, and pass that to upper layers?

File: drivers/ethernet/nxp_enet/eth_nxp_enet.c [hunk 2]
```diff
@@ -566,18 +572,48 @@ static void eth_nxp_enet_isr(const struct device *dev)
...
+#ifdef CONFIG_SOC_FAMILY_NXP_IMXRT
+	uint32_t mac_addr_fuse[2] = {0};
+
+	OCOTP_ReadFuseShadowRegisterExt((OCOTP_Type *)OCOTP_BASE,
+#if defined(CONFIG_SOC_SERIES_IMXRT10XX)
+		0x620,
+#elif defined(CONFIG_SOC_SERIES_IMXRT11XX)
+		0xA90,
+#endif
+		mac_addr_fuse, 2);
```
Comment (pdgendt) [inline drivers/ethernet/nxp_enet/eth_nxp_enet.c:604]: Putting these SoC specific code in a driver feels hacky. Maybe use a `syscon` driver layer?

Comment (decsny) [reply drivers/ethernet/nxp_enet/eth_nxp_enet.c:604]: that's actually what I was originally planning to do but there's not a valid enough reason to do that effort yet, and I am busy, but yes eventually the OCOTP should be a syscon driver IMO

File: drivers/ethernet/nxp_enet/eth_nxp_enet.c (hunk 1 above)
Comment (decsny) [reply drivers/ethernet/nxp_enet/eth_nxp_enet.c:593]: maybe instead of this I will add a build assert

**Review by dleach02 — DISMISSED** [review]

**Review by pdgendt — DISMISSED** [review]

**Review by pdgendt — APPROVED** [review]

**Review by danieldegrasse — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Pull request merged**